instagram-app/src/components/CommentSection/CommentSection.jsx

```diff
@@ -1,19 +1,82 @@
-import React from 'react';
-import Comment from './Comment';
-import './CommentSection.css';
-
-let CommentSection = ({ comments }) => {
-  return (
-    <div className="comment-section">
-      {comments.map((comment, index) => {
-        return(
-          <div key={index}>
-            <Comment comment={comment} />
-          </div>
-        );
-      })}
-    </div>
-  )
+import React, { useState, useEffect } from "react";
+import Comment from "./Comment";
+import { Input } from "reactstrap";
+import "./CommentSection.css";
+
+let CommentSection = ({ parentId, comments, timeAgo }) => {
+  const savedItem = JSON.parse(
+    window.localStorage.getItem("comments" + parentId)
+  );
+  const toInitialize = savedItem ? savedItem : comments;
+  const [storedComments, setStoredComments] = useState(toInitialize);
+  const [commentFormInput, setCommentFormInput] = useState("");
+
+  const saveToStorage = () => {
+    window.localStorage.setItem(
+      "comments" + parentId,
+      JSON.stringify(storedComments)
+    );
+    console.log(JSON.parse(window.localStorage.getItem("comments" + parentId)));
+  };
+
+  const addnewComment = () => {
+    const idLastCommentNumber = Number(
+      storedComments[storedComments.length - 1].id
+    );
+    let newComment = {
+      id: String(idLastCommentNumber + 1),
+      username: "arandomuser",
+      text: commentFormInput
+    };
+
+    setStoredComments(storedComments.concat(newComment));
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    addnewComment();
+    setCommentFormInput("");
+  };
+
+  const handleChange = event => {
+    setCommentFormInput(event.target.value);
+  };
+
+  useEffect(() => {
+    saveToStorage();
+  }, [storedComments]);
+
+  if (storedComments) {
+    return (
+      <>
+        <div className="comment-section">
+          {storedComments.map((comment, index) => {
+            return (
```

Comment:
I'm not a fan on how you decided to structure your code here. 
As a matter of personal preference I fill this component could be broken in multiple components. It would be more reusable and more easy to read and maintain.